lua/plenary/strings.lua

```diff
@@ -0,0 +1,117 @@
+local path = require('plenary.path').path
+
+local M = {}
+
+M.strdisplaywidth = (function()
+  if jit and path.sep ~= [[\]] then
+    local ffi = require('ffi')
+    ffi.cdef[[
+      typedef unsigned char char_u;
+      int linetabsize_col(int startcol, char_u *s);
+    ]]
+
+    return function(str, col)
+      local startcol = col or 0
+      local s = ffi.new('char[?]', #str + 1)
+      ffi.copy(s, str)
+      return ffi.C.linetabsize_col(startcol, s) - startcol
+    end
+  else
+    return function(str, col)
+      return #str - (col or 0)
```

Comment:
I just realized we could do something like as well here:
```lua
if not vim.in_fast_loop() then
  -- viml
else
  return #str - (col or 0)
end
```

Reply:
Good catch. I'll add it to other funcs.

Reply:
Add falbacks a3dce21